### PR TITLE
First prototype of animated background and logo on log in

### PR DESCRIPTION
**Pull request by jancborchardt**
Now this is only a first prototype to see if this would work. Ever so slightly nice animated background, and animated logo just like on the website. Please review @karlitschek @skjnldsv @MorrisJobke @pixelipo 

Things left to do:
- [ ] Adjust the background image so it’s repeatable (basically cut something off the bottom right and adjust the height so it links)
- [ ] Adjust the size of the logo so it’s not blurry
- [ ] Check all occurences of the logo to make it work there too

### CODE REVIEW

**Comment by karlitschek** [comment]
I like it. Nice subtil animation. Two small issues. The animated logo shows up fine on Chrome but not on Safari. Not sure why. And the animated background picture looks not totally balanced with some empty spots and some totally full spots. 
But in general super awesome!!

**Comment by karlitschek** [comment]
@jancborchardt

File: core/css/guest.css [hunk 1]
```diff
@@ -24,11 +24,25 @@ body {
...
 	background-color: #0082c9;
 	background-image: url('../img/background.png?v=2');
-	background-position: 50% 50%;
-	background-repeat: no-repeat;
+	background-position: 960px 0;
+	background-repeat: repeat;
 	background-size: cover;
 	background-attachment: fixed; /* fix background gradient */
 	height: 100%; /* fix sticky footer */
+	-webkit-animation: background-animation 240s linear 0s normal none infinite;
```
Comment (pixelipo) [inline core/css/guest.css]: @jancborchardt two issues with this:
1) vendor prefixes should **never** be used in production-level software. They were introduced as means to test work-in-progress CSS specifications.
2) `animation` spec has been finalized and implemented by all web browsers (IE9 and lower don't support it either way), so prefix usage in this case is deprecated

File: core/css/guest.css [hunk 2]
```diff
@@ -24,11 +24,25 @@ body {
...
+	-moz-animation: background-animation 240s linear 0s normal none infinite;
+	-ms-animation: background-animation 240s linear 0s normal none infinite;
+	-o-animation: background-animation 240s linear 0s normal none infinite;
+	animation: background-animation 200s linear 0s normal none infinite;
+}
+
+@-webkit-keyframes background-animation {
```
Comment (pixelipo) [inline core/css/guest.css]: Same as above, with that difference that this one is required for Android browser <5

File: core/css/guest.css [hunk 3]
```diff
@@ -75,6 +88,31 @@ h3 {
...
 
+#header .logo {
+	background: url('../img/logo-sprite.png') 0 0 no-repeat;
+	height: 100px;
+	width: 140px;
+	display: inline-block;
+	/*-webkit-transform: scale(.5);
+	zoom: .5;*/
+	background-position-x: 0;
+	-webkit-animation: logo-animation 2s steps(33) forwards;
```
Comment (pixelipo) [inline core/css/guest.css]: same as above, @jancborchardt

**Comment by MorrisJobke** [comment]
Just to let you know: this causes high CPU load in Chrome and Safari for me. And it's not because I use low end hardware (MacBook Pro from Mid 2015 with an i7)

**Review by pixelipo — CHANGES_REQUESTED** [review]
Please remove vendor prefixes

**Comment by jancborchardt** [comment]
@MorrisJobke oh, that's a killer. Can you isolate if it's the background movement or the logo animation?

**Comment by skjnldsv** [comment]
It's definitely the background, I go from 7% to 45% when activated. i5@desktop.

**Comment by jancborchardt** [comment]
Ouf, ok then we should absolutely not do that, not worth it. I thought CSS animations were supposed to be more performant?

Or does anyone know how we could do it differently.

**Comment by skjnldsv** [comment]
Seems to be relative to the transition you used: https://stackoverflow.com/a/35908679/3885878
Could you try with an image in a separate div maybe?

**Comment by jancborchardt** [comment]
@skjnldsv according to that post, with Gecko (Firefox) and Webkit (Safari) the issue would still be there cause it’s quite a long animation. :\

**Comment by skjnldsv** [comment]
@jancborchardt Anyway, I'm not quite sure an animation like that is really necessary.
Imho, I like the logo identity, but this feels very heavy and I rather have my fixed bg than this. :)

capture d ecran_2017-10-26_16-07-44

**Comment by pixelipo** [comment]
It's a very nice animation, but I think login page should be as lean as possible. I propose we scrap this until better times (faster CSS animations and/or SVG animation)

**Comment by jancborchardt** [comment]
I separated out the log in logo animation at https://github.com/nextcloud/server/pull/7050 because the background animation is too much performance penalty at little gain. The log in page is not visible for long anyway.

**Comment by Espina2** [comment]
You can animate css properties that make the whole layout paint again. When you want to do css animations always use transform properties. They use graphic acceleration, and don't need to apply a paint on every single element.

You can take a look on browser rendering optimization free course by Google.

**Comment by Espina2** [comment]
BTW can you please provide a gif or a video in what you did? 

Suggestion: If you want more feedbacks you should always include a before and after of the whole page, if is some interaction include a small video to show what you did. I sometimes have 5/10 minutes to see all the github notifications, Im seeing on my phone and I can go to a local installation switch the branch and take a look. This is very time consuming... So, if we do that, its clearly more easy and straight forward to everyone give feedback.

**Comment by jancborchardt** [comment]
Continued in https://github.com/nextcloud/server/pull/7050

**Pull request closed without merging**